kobo/apps/openrosa/apps/logger/models/instance.py

```diff
@@ -89,6 +90,7 @@ class Instance(AbstractTimeStampedModel):
     # do not use it anymore.
     deleted_at = models.DateTimeField(null=True, default=None)
 
+    root_uuid = models.CharField(max_length=249, null=True, unique=True, db_index=True)
```

Comment:
As per our discussion, let's using a unique together constraint with `root_uuid` and `xform_id`.